aten/src/ATen/TensorIterator.cpp

```diff
@@ -941,6 +945,9 @@ TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tenso
 void TensorIteratorBase::populate_operands(TensorIteratorConfig& config) {
   for (int i = 0; i < config.tensors_.size(); i++) {
     operands_.emplace_back(std::move(config.tensors_[i]));
+    if (operands_[i].tensor.is_meta()) {
+      is_meta_ = true;
```

Comment:
seems like this logic silently echoes emergent behavior of the current multidispatch algorithm + dispatch key declaration order, might be worth a comment